src/main.ts

```diff
@@ -1,37 +1,96 @@
 import * as temp from 'temp'
+import * as os from 'os'
 
 import * as artifact from '@actions/artifact'
 import * as core from '@actions/core'
 import * as exec from '@actions/exec'
 import * as glob from '@actions/glob'
 import * as io from '@actions/io'
 
-async function main(): Promise<void> {
-  try {
-    const homeDir: string | undefined = process.env.HOME
-    const buildDir = `${homeDir}/built-package`
+// Anything we pass as the optional 3rd param to exec.exec must implement the
+// ExecOptions interface here https://github.com/actions/toolkit/blob/master/packages/exec/src/interfaces.ts.
+// The only piece of that we actually need is the listeners, this class exists to
+// give us that.
+class ExecOptions {
+  public listeners: object = {}
+}
+
+// Maybe (if we can get it to work) we have a param for tacking on a new error
+// message
+async function execWrapper(commandLine: string,
+                           args?: string[]): Promise<number> {
+    let myOutput = ''
+    let myError = ''
+
+    let options = new ExecOptions
+    options.listeners = {
+      stdout: (data: Buffer) => {
+        myOutput += data.toString()
+      },
+      stderr: (data: Buffer) => {
+        myError += data.toString()
+      }
+    }
+
+    return await exec.exec(commandLine, args, options)
+}
+
+async function getCondaURL(): Promise<string> {
```

Comment:
Looks good, but I don't think this function is asynchronous, the type signature probably needs to be tweaked a bit to not return a promise, otherwise, this all is great!